### DISCUSSION TITLE
为什么FTP代理是灰色的？

### DISCUSSION THREAD

**Discussion by GuihongWang** [SeaHOH/GotoX#204] [问答/咨询] [answered]
![image](https://user-images.githubusercontent.com/57795156/129447415-d880aff1-78ad-4304-b348-ca3716371f83.png)
&
控制台一直在输出
[picker gws] 3 测试失败（超时：2000 ms）2a00:1450:401b:800::2017，(-1, '谷歌域名没有获取到正确的证书链：CA 公钥不匹配。')
即便我没有使用gws

**Comment by GuihongWang** [comment]
更新直连IP库的时候 出现错误（浏览器可以打开 但是gotox不能直接下载）
![image](https://user-images.githubusercontent.com/57795156/129447479-a17372ed-3419-48b7-8018-a8dc5ed8f299.png)
请求网址 'https://raw.githubusercontent.com/17mon/china_ip_list/master/china_ip_list.txt' 时，重试 2 次后仍然失败。

**Comment by SeaHOH** [comment, marked as answer]
- 灰色菜单
  这个代理设置是 GotoX 自用的，仅支持 HTTP/HTTPS 代理，其余两种是出于兼容考虑，设置 GotoX 代理时不覆盖这两种代理。  
  灰色代表系统中现在未设置有 FTP 代理，所有无法再“**禁用**”。

- CA 公钥不匹配
  参见 #197 ，需要升级版本。

- 更新直连IP库的时候出现错误
  更新时强制使用 GotoX 且请求的是未压缩文件，受网络故障影响较大，可以把 raw.githubusercontent.com 加入 CFW 代理规则使更新更加流畅。

**Reply by GuihongWang** [reply]
怎么添加进CFW代理规则？

**Reply by SeaHOH** [reply]
参见 config 文件夹 ActionFilter.ini 的注释详解，参考自带示例规则。
